evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -191,3 +228,159 @@ class Pipeline(pipeline_class):
+def test_classification_pipeline_encodes_targets(mock_score, mock_predict, mock_fit, pipeline_class, X_y_binary):
+    X, y = X_y_binary
+    y_series = pd.Series(y)
+    mock_predict.return_value = y_series
+    X = pd.DataFrame({"feature": range(len(y))})
+    y_encoded = y_series.map(lambda label: "positive" if label == 1 else "negative")
+
+    class MyTsPipeline(pipeline_class):
+        component_graph = ['Delayed Feature Transformer', 'Logistic Regression Classifier']
+
+    pl = MyTsPipeline({"Delayed Feature Transformer": {"gap": 0, "max_delay": 1},
+                       "pipeline": {"gap": 0, "max_delay": 1}})
+
+    pl.fit(X, y_encoded)
+
+    answer = pd.DataFrame({"feature": X.feature,
+                           "feature_delay_1": X.feature.shift(1),
+                           "target_delay_1": y_series.shift(1)}).dropna(axis=0, how='any')
+
+    df_passed_to_estimator, target_passed_to_estimator = mock_fit.call_args[0]
+
+    # Check the features have target values encoded as ints.
+    pd.testing.assert_frame_equal(df_passed_to_estimator, answer)
+
+    # Check that target is converted to ints. Use .iloc[1:] because the first feature row has NaNs
+    pd.testing.assert_series_equal(target_passed_to_estimator, y_series.iloc[1:])
+
+    pl.predict(X, y_encoded)
+    # Best way to get the argument since the api changes between 3.6/3.7 and 3.8
+    df_passed_to_predict = mock_predict.call_args[0][0]
+    pd.testing.assert_frame_equal(df_passed_to_predict, answer)
+
+    mock_predict.reset_mock()
+    # Since we mock score_all_objectives, the objective doesn't matter
+    pl.score(X, y_encoded, objectives=['MCC Binary'])
+    df_passed_to_predict = mock_predict.call_args[0][0]
+    pd.testing.assert_frame_equal(df_passed_to_predict, answer)
+
+
+@pytest.mark.parametrize("pipeline_class,objectives", [(TimeSeriesBinaryClassificationPipeline, ["MCC Binary"]),
+                                                       (TimeSeriesBinaryClassificationPipeline, ["Log Loss Binary"]),
+                                                       (TimeSeriesBinaryClassificationPipeline, ["MCC Binary", "Log Loss Binary"]),
+                                                       (TimeSeriesMulticlassClassificationPipeline, ["MCC Multiclass"]),
+                                                       (TimeSeriesMulticlassClassificationPipeline, ["Log Loss Multiclass"]),
+                                                       (TimeSeriesMulticlassClassificationPipeline, ["MCC Multiclass", "Log Loss Multiclass"]),
+                                                       (TimeSeriesRegressionPipeline, ['R2']),
+                                                       (TimeSeriesRegressionPipeline, ['R2', "Mean Absolute Percentage Error"])])
+@pytest.mark.parametrize("use_ww", [True, False])
+def test_score_works(pipeline_class, objectives, use_ww, X_y_binary, X_y_multi, X_y_regression):
```

Comment:
I think it's good to have a unit test where we pass data through a minimal time series pipeline and make sure everything works. I'm setting n_jobs=1 here and the test takes 2.5 seconds to run so I think it's ok.